src/containers/articles/ShareArticle.js

```diff
@@ -0,0 +1,125 @@
+import React, { Component } from "react";
+import { apiCalls } from "../../Helpers/axios";
+import Input from "../../components/Input/Input";
+import Modal from "../../components/Modal/Modal";
+export class ShareBar extends Component {
+  state = {
+    openmodal: false,
+    email: ""
+  };
+  handleSuccessShareResponse = ({ data }) => {
+    const newLink = data.shared_link.replace(
+      "http://ah-premier-staging.herokuapp.com/api",
+      "https://ah-fortem-staging.herokuapp.com"
+    );
+    const newWindow = window.open(newLink, "_blank");
+    newWindow.focus();
+  };
+
+  handleEmailChange = event => {
+    this.setState({
+      email: event.target.value
+    });
+  };
+
+  handleShareSocial = (slug, platform, data = {}) => () => {
+    const token = localStorage.getItem("token");
+    return apiCalls("post", "/articles/" + slug + "/share/" + platform, data, {
+      Authorization: `Bearer ` + token
+    }).then(this.handleSuccessShareResponse);
```

Comment:
Nice work, I like the way we get redirected to a new page to share the article, however, if I am not logged in the article does not get shared and I get no message on the same. Maybe a catch can help here

```js

.catch(() => {
ShowMessage({
message: "Log in to share the article ...",
type: "error"
})
})
```

Reply:
Thanks for catching this. I have now added a catch and the error message is displayed in our reusable snackbar